### PR TITLE
Proof of concept: float string field

### PR DESCRIPTION
**Pull request by mllocs**
Tries to fix this issue: https://github.com/openfoodfoundation/openfoodnetwork/issues/853

I'm basically using a hidden input field to store the numeric conversion of the input text field. Whenever the user types a value in the visible input, it is transformed and stored in the hidden field as a numeric value. 

http://g.recordit.co/kzHcqGD5qg.gif

This is just a proof of concept, applied to the `price` attribute in the "New product" form. 

I added a partial view that adds the hidden field and the javascript that syncs both fields, and then a Rails helper to make it easier to use that partial. 

---

Note:

With this fix, we accept any kind of format as an input but **we always display those values in the standard numeric way**, which means that it uses the dot for decimals and no thousands separator.

### CODE REVIEW

File: app/assets/javascripts/shared/util.js.coffee [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  # If string ends in a single digit (e.g. ,2) make it ,20 in order for the result to be in "cents"
+  if string.match(/[\.,]\d$/)
+    string = string + '0'
+
+  # If does not end in ,00 / .00 then add trailing 00 to turn it into cents
+  if !string.match(/[\.,]\d\d$/)
+    string = string + '00'
+
+  # Replace all Currency Symbols, Letters and -- from the string
```
Comment (sauloperez) [inline app/assets/javascripts/shared/util.js.coffee]: I'd say "remove". That's what we do, right?

**Comment by myriamboure** [comment]
Sounds good! I think using a standard display with dot as decimal is really ok, as long as people can fill in either with comas or dots or any other conventionnal way ;-) @ltrls what do you think of this proposal ?

**Comment by ltrls** [comment]
That's smart!
> With this fix, we accept any kind of format as an input but we always display those values in the standard numeric way, which means that it uses the dot for decimals and no thousands separator.

Price amounts are injected as localized strings (using Money#to_s). It's causing the #1623 issue as the [LocalizeCurrency](https://github.com/openfoodfoundation/openfoodnetwork/blob/0760d4cc1f26fd06f60b420293e3671f205d0a07/app/assets/javascripts/darkswarm/filters/localize_currency.js.coffee) filter fails at parsing the string if it's not in English format. I'm thinking of serializing amounts in English format and improving the filter to localize the separator too (as it's written at L11, but probably injecting the locale instead of the separator and using [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString)), so that should solves this too :)

**Comment by oeoeaio** [comment]
Others are better placed that me to assess the functionality here, and I realise that this is a spike, so is not fully formed, but just wanted to flag now that I would love to see this written as an angular directive so that we aren't polluting the global namespace. Specs etc. too, ya ya ya.

**Comment by mkllnk** [comment]
Are we doing this in Javascript so that we don't have to deal with Spree? I had a Rails solution in mind. For example in `app/controllers/spree/admin/products_controller_decorator.rb`:
```ruby
  before_filter :parse_price

  def parse_price
    params[:product][:price] = stringToPrice params[:product][:price]
  end
```

There is probably a better way to organise the code and have it in a nice little lib file.
Do you see any downsides of that approach?

**Comment by daniellemoorhead** [comment]
@ltrls @oeoeaio @mkllnk can we close this spike PR now that #1886 is merged?

**Comment by mkllnk** [comment]
@daniellemoorhead #853 is still not solved. #1886 solved #1623. I think that the competing pull request to this one is #1831 which is very far, but not merged yet. I do prefer #1831 and would vote for closing this spike even though it is very interesting.

**Comment by daniellemoorhead** [comment]
Closing in preference for #1831. Feel free to reopen this if there is value in keeping it open for now :)

**Pull request closed without merging**